src/Editor/GUI/Editors/CurveEditor.cpp

```diff
@@ -0,0 +1,740 @@
+            ImVec2 b = points[i];
+            a.y = 1 - a.y;
+            b.y = 1 - b.y;
+            a = a * (bb.Max - bb.Min) + bb.Min;
+            b = b * (bb.Max - bb.Min) + bb.Min;
+            window->DrawList->AddLine(a, b, GetColorU32(ImGuiCol_PlotLinesHovered));
+        }
+
+        if (hovered) {
+            // control points
+            for (i = 0; i < max; i++) {
+                ImVec2 p = points[i];
+                p.y = 1 - p.y;
+                p = p * (bb.Max - bb.Min) + bb.Min;
+                ImVec2 a = p - ImVec2(2, 2);
+                ImVec2 b = p + ImVec2(2, 2);
+                window->DrawList->AddRect(a, b, GetColorU32(ImGuiCol_PlotLinesHovered));
+            }
+        }
+
+        // buttons; @todo: mirror, smooth, tessellate
+        if (ImGui::Button("Flip")) {
+            for (i = 0; i < max; ++i) {
+                points[i].y = 1 - points[i].y;
+            }
+        }
+        ImGui::SameLine();
+
+        // curve selector
+        const char* items[] = {
+            "Custom",
+
+            "Linear",
+            "Quad in",
+            "Quad out",
+            "Quad in  out",
+            "Cubic in",
+            "Cubic out",
+            "Cubic in  out",
+            "Quart in",
+            "Quart out",
+            "Quart in  out",
+            "Quint in",
+            "Quint out",
+            "Quint in  out",
+            "Sine in",
+            "Sine out",
+            "Sine in  out",
+            "Expo in",
+            "Expo out",
+            "Expo in  out",
+            "Circ in",
+            "Circ out",
+            "Circ in  out",
+            "Elastic in",
+            "Elastic out",
+            "Elastic in  out",
+            "Back in",
+            "Back out",
+            "Back in  out",
+            "Bounce in",
+            "Bounce out",
+            "Bounce in out",
+
+            "Sine square",
+            "Exponential",
+
+            "Schubring1",
+            "Schubring2",
+            "Schubring3",
+
+            "SinPi2",
+            "Swing"
+        };
+        if (modified) {
+            item = 0;
+        }
+        if (ImGui::Combo("Ease type", &item, items, IM_ARRAYSIZE(items))) {
+            max = maxpoints;
+            if (item > 0) {
+                for (i = 0; i < max; ++i) {
+                    points[i].x = i / float(max - 1);
+                    points[i].y = float(tween::ease(item - 1, points[i].x));
+                }
+            }
+        }
+
+        char buf[128];
+        const char *str = label;
+
+        if (hovered) {
+            ImVec2 pos = (g.IO.MousePos - bb.Min) / (bb.Max - bb.Min);
+            pos.y = 1 - pos.y;
+
+            sprintf(buf, "%s (%f,%f)", label, pos.x, pos.y);
+            str = buf;
+        }
+
+        RenderTextClipped(ImVec2(bb.Min.x, bb.Min.y + style.FramePadding.y), bb.Max, str, NULL, NULL, ImVec2(100, 100));
+
+        return modified;
+    }
+
+};
+
+CurveEditor::CurveEditor() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Member variable 'CurveEditor::curveBuf' is not initialized in the constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)